compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt

```diff
@@ -164,6 +165,7 @@ fun IrClass.addFunction(
         this.origin = origin
     }.apply {
         if (!isStatic) {
+            copyTypeParametersFrom(parentAsClass)
```

Comment:
without this I got exception later in _remapTypeParameters_

Reply:
This change doesn't look correct. Could you clarify what exception were you getting?

Reply:
```
java.lang.IndexOutOfBoundsException: Empty list doesn't contain element at index 0.
	at kotlin.collections.EmptyList.get(Collections.kt:36)
	at kotlin.collections.EmptyList.get(Collections.kt:24)
	at org.jetbrains.kotlin.backend.common.ir.IrUtilsKt.remapTypeParameters(IrUtils.kt:328)
	at org.jetbrains.kotlin.backend.common.ir.IrUtilsKt.remapTypeParameters(IrUtils.kt:341)
	at org.jetbrains.kotlin.backend.common.ir.IrUtilsKt.copyTo$default(IrUtils.kt:119)
	at org.jetbrains.kotlin.ir.builders.declarations.DeclarationBuildersKt.addFunction(declarationBuilders.kt:169)
	at org.jetbrains.kotlin.ir.builders.declarations.DeclarationBuildersKt.addFunction$default(declarationBuilders.kt:144)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.addOverride(ParcelizeIrTransformer.kt:279)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.visitClass(ParcelizeIrTransformer.kt:115)
	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitClass(IrElementVisitorVoid.kt:44)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.visitClass(ParcelizeIrTransformer.kt:42)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.visitClass(ParcelizeIrTransformer.kt:42)
```

Reply:
btw what is the right place for more 'unit' tests for IR builders?

Reply:
replaced with explicit passing type argument to copyTo